server/swap/swap.go

```diff
@@ -1475,15 +1474,16 @@ func (s *Swapper) processAck(msg *msgjson.Message, acker *messageAcker) {
 	if !acker.isMaker {
 		acker.match.Sigs.TakerRedeem = ack.Sig
 		if err = s.storage.SaveRedeemAckSigB(mktMatch, ack.Sig); err != nil {
-			s.respondError(msg.ID, acker.user, msgjson.UnknownMarketError, "internal server error")
+			s.respondError(msg.ID, acker.user, msgjson.UnknownMarketError,
+				fmt.Sprintf("internal server error: %v", err))
```

Comment:
For any of the errors from an Archivist, I'm inclined to log server-side and omit the actual error from the response to client.  Unless we are explicitly checking the error type (see the `Is*` helpers in `server/db/errors.go`), we should assume the error is an internal failure and that's all the client needs to know.